interface-impl-amf/src/main/java/org/mule/amf/impl/loader/ProvidedResourceLoader.java

```diff
@@ -31,22 +34,58 @@ public CompletableFuture<Content> fetch(String resourceName) {
       throw new RuntimeException("Failed to apply.");
     }
 
-    final URI resource = resourceLoader.getResource(resourceName);
+    try {
+      InputStream streamResource = getResourceFromURI(resourceName);
 
-    if (resource != null) {
-      final InputStream stream = resourceLoader.getResourceAsStream(resourceName);
-      try {
-        final String resourceAsString = IOUtils.toString(stream);
-        final Content content = new Content(resourceAsString, resourceName);
-        future.complete(content);
-      } catch (IOException e) {
-        future.completeExceptionally(new RuntimeException("Failed to fetch resource '" + resourceName + "'"));
+      if (streamResource != null) {
+        future.complete(new Content(getContentFromStream(streamResource), resourceName));
+        return future;
+      }
+
+      URI resourceUri = resourceLoader.getResource(resourceName);
+
+      if (resourceUri != null) {
+        streamResource = resourceLoader.getResourceAsStream(resourceName);
+        future.complete(new Content(getContentFromStream(streamResource), resourceUri.toString()));
       }
-    } else {
       future.completeExceptionally(new Exception("Failed to fetch resource '" + resourceName + "'"));
+      return future;
+
+    } catch (Exception e) {
+      future.completeExceptionally(new RuntimeException("Failed to fetch resource '" + resourceName + "'"));
```

Comment:
```suggestion
      future.completeExceptionally(new RuntimeException("Failed to fetch resource '" + resourceName + "'", e));
```

Reply:
this was just for keeping the same error